### PR TITLE
added: allow specifying a function for beta

### PR DESCRIPTION
**Pull request by akva2**
t holds the adaptive step number. this allows for
flexible specification of how much to refine at each step.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/SIM/AdaptiveSetup.h [hunk 1]
```diff
@@ -96,6 +97,7 @@ class AdaptiveSetup
   bool   alone;      //!< If \e false, this class is wrapped by SIMSolver
   bool   linIndep;   //!< Test mesh for linear independence after refinement
   double beta;       //!< Refinement percentage in each step
+  RealFunc* betaFunc;//!< Beta as a function
```
Comment (kmokstad) [inline src/SIM/AdaptiveSetup.h:100]: Suggest using `ScalarFunc` instead, since `RealFunc` is a spatial function.

File: src/SIM/AdaptiveSetup.h [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 #include "MatVec.h"
 
+class RealFunc;
```
Comment (kmokstad) [inline src/SIM/AdaptiveSetup.h:19]: ScalarFunc

File: src/SIM/AdaptiveSetup.C [hunk 3]
```diff
@@ -272,7 +285,13 @@ int AdaptiveSetup::calcRefinement (LR::RefineData& prm, int iStep,
...
   prm.options.reserve(8);
-  prm.options.push_back(beta);
+  if (betaFunc) {
+    Vec4 X;
+    X.t = iStep-1;
+    prm.options.push_back((*betaFunc)(X));
+  } else
+    prm.options.push_back(beta);
+
   prm.options.push_back(knot_mult);
```
Comment (kmokstad) [inline src/SIM/AdaptiveSetup.C:295]: If using `ScalarFunc` instead, then L288-295 could be replaced by the single line:
```
  prm.options.push_back(betaFunc ? (*betaFunc)(iStep-1) : beta);
```

**Review by kmokstad — CHANGES_REQUESTED** [review]
Using ScalarFunc instead would simplify

File: src/SIM/AdaptiveSetup.h (hunk 1 above)
Comment (akva2) [reply src/SIM/AdaptiveSetup.h]: okay. it's a bit iffy with the printing going inside the parsing function but i think i got something that is okay.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Pull request merged**